### PR TITLE
Fixed issue #17276: 500: Internal Server Error - Call to a member function format() on bool

### PR DESCRIPTION
**Pull request by gabrieljenik**
- Revert "Dev Removed obsolete Date_Time_Converter helper class" and fix conflicts
- This reverts commit 0f8dcc2ec477125117c775db6a63bc5bd110683f.

### CODE REVIEW

File: application/controllers/SurveyAdministrationController.php [hunk 1]
```diff
@@ -2309,11 +2309,13 @@ public function actionExpireMultipleSurveys()
...
         $expires = App()->request->getPost('expires');
         $formatdata = getDateFormatData(Yii::app()->session['dateformat']);
+        Yii::import('application.libraries.Date_Time_Converter', true);
         if (trim($expires) == "") {
             $expires = null;
         } else {
-            $datetimeobj = DateTime::createFromFormat($formatdata['phpdate'] . ' H:i', $expires);
-            $expires = $datetimeobj->format("Y-m-d H:i:s");
+            //new Date_Time_Converter($expires, $formatdata['phpdate'].' H:i');
+            $datetimeobj = new date_time_converter($expires, $formatdata['phpdate'] . ' H:i');
```
Comment (olleharstedt) [inline application/controllers/SurveyAdministrationController.php:2317]: Class spelling is different, but maybe it doesn't matter? date_time_converter vs Date_Time_Converter.

**Comment by olleharstedt** [comment]
Was this tested on PHP 8 too?

**Comment by gabrieljenik** [comment]
> Was this tested on PHP 8 too?

Yes, we did a basic text on PHP8. Haven't tested all revertion points, just a few.

**Comment by olleharstedt** [comment]
Great!

File: application/controllers/SurveyAdministrationController.php (hunk 1 above)
Comment (olleharstedt) [reply application/controllers/SurveyAdministrationController.php:2317]: https://stackoverflow.com/questions/5260168/capital-letters-in-class-name-php

Comment (olleharstedt) [reply application/controllers/SurveyAdministrationController.php:2317]: Not case sensitive. But still.

Comment (gabrieljenik) [reply application/controllers/SurveyAdministrationController.php:2317]: Let us make things straight, just in case

Comment (Shnoulle) [reply application/controllers/SurveyAdministrationController.php:2317]: Autoloading on linux can broke (but no autoload here)

**Pull request merged**